src/driver_rigel.cpp

```diff
@@ -138,12 +138,21 @@ class PropertyDriverRigel : public IPropertyDriver {
       case CAP_PROP_LEAP_LEDS:
         *max_val = 1;
         break;
+      case CAP_PROP_LEAP_PULSEWIDTHMODULATION: {
+        vector<uint8_t> buf(sizeof(LEAP_DEVCONFIG));
+        ok = dev_->get_xu(leap_xu_, LEAP_XU_DEVCONFIG, buf.data(), sizeof(LEAP_DEVCONFIG));

```

Comment:
Can you add a check for get_xu's return value before using buf.data() ?

Reply:
Sure, that's now done :)

Reply:
get_prop_range could now throw : I let you judge if 'get_xu' should be in a try-catch

Reply:
That's a good thing to flag up but I don't think it's an issue. Not all backend have a `get_xu` that those, and for those that do, it's due to `ioctl` failure or something else low-level. These are exceptional cases and given that this is C++ all the way up, I don't think it's right to handle the exception here.